### ISSUE TITLE
Bricks not comming back online after reboot

### ISSUE THREAD

**Issue by falencastro** [openshift/openshift-ansible#11229]
#### Description

After I restart gluster pods and nodes, bricks don't come back online.

Issue on upstream: [499](https://github.com/gluster/gluster-kubernetes/issues/499)

Commit [f2bd499](https://github.com/openshift/openshift-ansible/commit/f2bd499c230b60dd7efa97372efb317bd5495c9f#diff-21b02900b0aa9e6d4b9b771d903e1202) re-added /dev bind-mount, but /dev/mapper inside the pod is comming empty, thus preventing brick lvs from being mounted. 

##### Version

Please put the following version information in the code block
indicated below.

* Your ansible version per `ansible --version`

If you're operating from a **git clone**:

* The output of `git describe`

If you're running from playbooks installed via RPM

* The output of `rpm -q openshift-ansible`

Place the output between the code block below:

```
ansible 2.6.5
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/felipe_alencastro/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Jul 13 2018, 13:06:57) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]

openshift-ansible-3.11.87-1-6-gc0985f8
```

##### Steps To Reproduce
1. Restart storage node
2. Delete gluster pod

##### Expected Results
Bricks back online after reboot

##### Observed Results
Describe what is actually happening.

```
sh-4.2# /usr/sbin/gluster-setup.sh 
mount: special device /dev/mapper/vg_e35c49f0d2599b9f8ef30d91f51e178e-brick_7d0289f91a81c81da1d46de8df7f64f9 does not exist
mount: special device /dev/mapper/vg_e35c49f0d2599b9f8ef30d91f51e178e-brick_504d9fa2e327e0e8f1dc734f71c2a8f8 does not exist
mount: special device /dev/mapper/vg_eef15aafba3b635faabfb415c1ea09cf-brick_07adafbdb2b6935722bb4d5129c28af5 does not exist
mount: special device /dev/mapper/vg_924057fd1923154863680dce794a9d65-brick_63d5299115541e4f86cf017e534f906b does not exist
mount: special device /dev/mapper/vg_e35c49f0d2599b9f8ef30d91f51e178e-brick_7b30e62fa6ae6e0c21e25bf519cc9929 does not exist
mount: special device /dev/mapper/vg_85cf451c446c4f40c0137683d1e7263b-brick_e72b5ebee642da54e9b0f7e1c8c16f9a does not exist
mount: special device /dev/mapper/vg_5c28b9948b307309b430092927a3fee9-brick_81c392b8e3f0d98a4c71954b4d6dea52 does not exist
mount: special device /dev/mapper/vg_a8870153905957d11b9d3e6b5c5ea516-brick_4ee4c1aaeeaf9537afdd3a109b3b69c8 does not exist
mount: special device /dev/mapper/vg_b8b26efd57904a34c56b2a94aa95113d-brick_695dec2f0544afa57991bf0891010a10 does not exist
mount: special device /dev/mapper/vg_a8870153905957d11b9d3e6b5c5ea516-brick_96e5069eaafe87cebfaeb8df10c9f348 does not exist
mount: special device /dev/mapper/vg_b8b26efd57904a34c56b2a94aa95113d-brick_af9dc5184430d19fd4251f6152a108e4 does not exist
mount: special device /dev/mapper/vg_5c28b9948b307309b430092927a3fee9-brick_6185a462b46a32b1e481b0c47b421c30 does not exist
Mount Successful
/etc/glusterfs is not empty
/var/log/glusterfs is not empty
/var/lib/glusterd is not empty
```

##### Additional Information

Provide any additional information which may help us diagnose the
issue.

* CentOS 7.6

##### Workaround

Apply manually [d2bdffe](https://github.com/openshift/openshift-ansible/commit/d2bdffea98d4770bfda0a25415ecde9e8243e820#diff-21b02900b0aa9e6d4b9b771d903e1202), keeping both the /dev bind mount and /dev/mapper bind mount.

`...`
`- mountPath: /mnt/host-dev`
  `name: glusterfs-dev`
`- mountPath: /dev/mapper`
  `name: glusterfs-dev-mapper`
`...`
`- hostPath:`
    `path: /dev`
    `type: ""`
  `name: glusterfs-dev`
`- hostPath:`
    `path: /dev/mapper`
    `type: ""`
  `name: glusterfs-dev-mapper`
`...`

**Comment by jarrpa** [comment]
Which version of the Gluster container image are you running? This should be fixed in the latest images...

**Comment by jarrpa** [comment]
@nixpanic for awareness

**Comment by nixpanic** [comment]
This should have been fixed with gluster/gluster-containers#115.

**Comment by falencastro** [comment]
> Which version of the Gluster container image are you running? This should be fixed in the latest images...

I'm using docker.io/gluster/gluster-centos:gluster3u12_centos7 (I believe 3.12 is the right version for OKD 3.11, right?), which doesn't have /usr/local/bin/update-params.sh. then I tried docker.io/gluster/gluster-centos:gluster4u1_centos7 (built 4 days ago), which has /usr/local/bin/update-params.sh, but hasn't [#115 fix](https://github.com/gluster/gluster-containers/pull/115), then at last I tried docker.io/gluster/gluster-centos:latest, which has the fix, but seems to be incompatible with cri-o-1.11.7-1.rhaos3.11.git17bfe63.el7.x86_64:

`[felipe_alencastro@alencastro ~] $ oc rsh glusterfs-storage-j2whz
exec failed: container_linux.go:336: starting container process caused "read init-p: connection reset by peer"
command terminated with exit code 1`

Seems to me [#115](https://github.com/gluster/gluster-containers/pull/115), should be backported to 3.12

**Comment by nixpanic** [comment]
The Gluster version is independent of OKD or kubernetes. docker.io/gluster/gluster-centos:latest is expected to work with CRI-O, and the changes are actually done to support deploying on CRI-O.

The error message that you posted can be related to a `/dev` mount in the pod, CRI-O prevents this from being done. The `/dev` from the host should be mounted on `HOST_DEV_DIR` that is something else than `/dev` in the container (we use `/mnt/host-dev` by default). With the current templates in openshift-ansible-3.11 everything should work (latest Gluster container + CRI-O).

**Comment by falencastro** [comment]
> er version is independent of OKD or kubernetes. docker.io/gluster/gluster-centos:latest is expected to work with CRI-O, an

I thought 3.12 was the right version, as it's what openshift enterprise uses. About the error, yes I'm using the latest glusterfs-storage deamonset template, with the latest cri-o available for CentOS (cri-o-1.11.7-1.rhaos3.11.git17bfe63.el7.x86_64) with `HOST_DEV_DIR` set to `/mnt/host-dev` and gluster-centos:latest.

On my OpenShift Enterprise deployment I don't face this issue, and there are 2 main differences:

* Image registry.access.redhat.com/rhgs3/rhgs-server-rhel7:v3.11 uses gluster 3.12.2, with `HOST_DEV_DIR` and [#115](https://github.com/gluster/gluster-containers/pull/115)
* cri-o-1.11.11-1.rhaos3.11.git474f73d.el7.x86_64

**Comment by falencastro** [comment]
Ok, I got it working, but not with the default cri-o, had to use [cri-o-1.11.11-1.rhaos3.11.git474f73d.el7.x86_64.rpm](https://buildlogs.centos.org/centos/7/paas/x86_64/openshift-origin311/cri-o-1.11.11-1.rhaos3.11.git474f73d.el7.x86_64.rpm) from buildlogs.centos.org, which isn't the default repo, and docker.io/gluster/gluster-centos:latest image.

Two things would be nice: cri-o-1.11.11 being promoted to the default repo, would make everything work ootb, and [#115](https://github.com/gluster/gluster-containers/pull/115) being applied to the other non 'latest' images, would also make it work with the default cri-o version (1.11.7), also people might want to have tighter version control of the cluster by locking the daemonset to a version tag.
